### PR TITLE
Using kubectl instead of the API

### PR DESCRIPTION
**Pull request by Skarlso**
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by aclevername — APPROVED** [review]

File: integration/tests/karpenter/karpenter_test.go [hunk 1]
```diff
@@ -101,47 +97,17 @@ var _ = Describe("(Integration) Karpenter", func() {
...
-	}
-
-	for _, selector := range selectors {
-		pods := clientset.CoreV1().Pods(karpenter.DefaultNamespace)
-		ps, err := pods.List(context.Background(), metav1.ListOptions{
-			LabelSelector: selector,
-		})
+		fmt.Fprintf(GinkgoWriter, "describe replicaset %s", name)
+		fmt.Fprint(GinkgoWriter, string(output))
+		cmd = exec.Command("kubectl", "describe", "deployment", "name", "-n", karpenter.DefaultNamespace)
```
Comment (cPu1) [inline integration/tests/karpenter/karpenter_test.go:107]:
```suggestion
		cmd = exec.Command("kubectl", "describe", "deployment", name, "-n", karpenter.DefaultNamespace)
```

File: integration/tests/karpenter/karpenter_test.go [hunk 2]
```diff
@@ -101,47 +97,17 @@ var _ = Describe("(Integration) Karpenter", func() {
...
-	Expect(err).NotTo(HaveOccurred())
-	var webhookEvents []eventsv1.Event
-	webhookEvents = append(webhookEvents, wes.Items...)
-	for wes.Continue != "" {
-		wes, err = events.List(context.Background(), metav1.ListOptions{
-			Continue: wes.Continue,
-		})
+func describeKarpenterResources(names []string) {
+	for _, name := range names {
+		cmd := exec.Command("kubectl", "describe", "replicaset", "name", "-n", karpenter.DefaultNamespace)
```
Comment (cPu1) [inline integration/tests/karpenter/karpenter_test.go:102]: The backing ReplicaSet's name will not match the Deployment's name. You'll have to use a label selector to query the RS.

Comment (Skarlso) [reply integration/tests/karpenter/karpenter_test.go:102]: Nope, this works fine.

File: integration/tests/karpenter/karpenter_test.go (hunk 1 above)
Comment (Skarlso) [reply integration/tests/karpenter/karpenter_test.go:107]: ah good spot. :D

File: integration/tests/karpenter/karpenter_test.go (hunk 2 above)
Comment (Skarlso) [reply integration/tests/karpenter/karpenter_test.go:102]:
```
➜  Run integration tests k describe deployment -n karpenter karpenter-webhook
Name:               karpenter-webhook
Namespace:          karpenter
CreationTimestamp:  Mon, 17 Jan 2022 16:28:38 +0100
Labels:             app.kubernetes.io/managed-by=Helm
Annotations:        deployment.kubernetes.io/revision: 1
                    meta.helm.sh/release-name: karpenter
                    meta.helm.sh/release-namespace: karpenter
Selector:           karpenter=webhook
Replicas:           1 desired | 0 updated | 0 total | 0 available | 1 unavailable
StrategyType:       Recreate
MinReadySeconds:    0
Pod Template:
  Labels:           karpenter=webhook
  Service Account:  karpenter
  Containers:
   webhook:
    Image:      public.ecr.aws/karpenter/webhook:v0.4.3@sha256:576628e37932715cdef2d5bc4bc641a255803b81a2ffb104d674f9f996b10936
    Port:       8443/TCP
    Host Port:  0/TCP
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:     100m
      memory:  50Mi
    Liveness:  http-get https://:8443/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CLUSTER_NAME:
      CLUSTER_ENDPOINT:
      SYSTEM_NAMESPACE:   (v1:metadata.namespace)
    Mounts:              <none>
  Volumes:               <none>
  Priority Class Name:   system-cluster-critical
Conditions:
  Type             Status  Reason
  ----             ------  ------
  Available        False   MinimumReplicasUnavailable
  ReplicaFailure   True    FailedCreate
  Progressing      False   ProgressDeadlineExceeded
OldReplicaSets:    <none>
NewReplicaSet:     karpenter-webhook-64485c8bd4 (0/1 replicas created)
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  13m   deployment-controller  Scaled up replica set karpenter-webhook-64485c8bd4 to 1
➜  Run integration tests k describe replicaset -n karpenter karpenter-webhook
Name:           karpenter-webhook-64485c8bd4
Namespace:      karpenter
Selector:       karpenter=webhook,pod-template-hash=64485c8bd4
Labels:         karpenter=webhook
                pod-template-hash=64485c8bd4
Annotations:    deployment.kubernetes.io/desired-replicas: 1
                deployment.kubernetes.io/max-replicas: 1
                deployment.kubernetes.io/revision: 1
                meta.helm.sh/release-name: karpenter
                meta.helm.sh/release-namespace: karpenter
Controlled By:  Deployment/karpenter-webhook
Replicas:       0 current / 1 desired
Pods Status:    0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           karpenter=webhook
                    pod-template-hash=64485c8bd4
  Service Account:  karpenter
  Containers:
   webhook:
    Image:      public.ecr.aws/karpenter/webhook:v0.4.3@sha256:576628e37932715cdef2d5bc4bc641a255803b81a2ffb104d674f9f996b10936
    Port:       8443/TCP
    Host Port:  0/TCP
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:     100m
      memory:  50Mi
    Liveness:  http-get https://:8443/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CLUSTER_NAME:
      CLUSTER_ENDPOINT:
      SYSTEM_NAMESPACE:   (v1:metadata.namespace)
    Mounts:              <none>
  Volumes:               <none>
  Priority Class Name:   system-cluster-critical
Conditions:
  Type             Status  Reason
  ----             ------  ------
  ReplicaFailure   True    FailedCreate
Events:
  Type     Reason        Age                   From                   Message
  ----     ------        ----                  ----                   -------
  Warning  FailedCreate  2m26s (x18 over 13m)  replicaset-controller  Error creating: pods "karpenter-webhook-64485c8bd4-" is forbidden: error looking up service account karpenter/karpenter: serviceaccount "karpenter" not found

```

**Pull request merged**